library/bigip_software.py

```diff
@@ -212,673 +290,1060 @@
       hotfix: "/root/Hotfix-BIGIP-11.6.0.3.0.412-HF3.iso"
       reuse_inactive_volume: yes
       state: "activated"
+
+- name: Activate (download, install, reboot, reuse_inactive_volume) base image and hotfix
+  bigip_software:
+      server: "bigip.localhost.localdomain"
+      user: "admin"
+      password: "admin"
+      hotfix: "http://fake.com/Hotfix-12.1.2.1.0.271-HF1.iso"
+      hotfix_md5sum: "http://fake.com/Hotfix-12.1.2.1.0.271-HF1.iso.md5"
+      software: "http://fake.com/BIGIP-12.1.2.0.0.249.iso"
+      software_md5sum: "http://fake.com/BIGIP-12.1.2.0.0.249.iso.md5"
+      build: "1.0.271"
+      version: "12.1.2"
+      remote_src: "yes"
+      state: "activated"
+      reuse_inactive_volume: True
+
+- name: Download hotfix image
+  bigip_software:
+      server: "bigip.localhost.localdomain"
+      user: "admin"
+      password: "admin"
+      hotfix: "http://fake.com/Hotfix-12.1.2.1.0.271-HF1.iso"
+      hotfix_md5sum: "http://fake.com/Hotfix-12.1.2.1.0.271-HF1.iso.md5"
+      build: "1.0.271"
+      version: "12.1.2"
+      remote_src: "yes"
+      state: "present"
+
+- name: Remove uploaded hotfix image
+  bigip_software:
+      server: "bigip.localhost.localdomain"
+      user: "admin"
+      password: "admin"
+      hotfix: "http://fake.com/Hotfix-12.1.2.1.0.271-HF1.iso"
+      build: "1.0.271"
+      version: "12.1.2"
+      remote_src: "yes"
+
+- name: Install (download, install) base image
+  bigip_software:
+      server: "bigip.localhost.localdomain"
+      user: "admin"
+      password: "admin"
+      software: "http://fake.com/BIGIP-12.1.2.0.0.249.iso"
+      software_md5sum: "http://fake.com/BIGIP-12.1.2.0.0.249.iso.md5"
+      build: "0.0.249"
+      version: "12.1.2"
+      remote_src: "yes"
+      volume: "HD1.1"
+      state: "installed"
+
+- name: Install (download, install) base image and hotfix
+  bigip_software:
+      server: "bigip.localhost.localdomain"
+      user: "admin"
+      password: "admin"
+      hotfix: "http://fake.com/Hotfix-12.1.2.1.0.271-HF1.iso"
+      hotfix_md5sum: "http://fake.com/Hotfix-12.1.2.1.0.271-HF1.iso.md5"
+      software: "http://fake.com/BIGIP-12.1.2.0.0.249.iso"
+      software_md5sum: "http://fake.com/BIGIP-12.1.2.0.0.249.iso.md5"
+      build: "1.0.271"
+      version: "12.1.2"
+      remote_src: "yes"
+      state: "installed"
+      volume: "HD1.2"
 '''
 
```

Comment:
https://github.com/F5Networks/f5-ansible/blob/devel/library/bigip_pool.py#L185

you're missing the RETURN variable